### PR TITLE
Add `glibc` via `gcompat` layer

### PR DESCRIPTION
**Pull request by harryzcy**
# Proposed changes

Use `gcompat` layer to use add `glibc` to alpine

## Readiness checklist

In order to have this pull request merged, complete the following tasks.

### Pull request author tasks

- [x] I included all the needed documentation for this change.
- [x] I provided the necessary tests.
- [x] I squashed all the commits into a single commit.
- [x] I followed the [Conventional Commit v1.0.0 spec](https://www.conventionalcommits.org/en/v1.0.0/).
- [x] I wrote the necessary upgrade instructions in the [upgrade guide](../docs/upgrade-guide.md).

### Super-linter maintainer tasks

- [ ] Label as `breaking` if this change breaks compatibility with the previous released version.
- [ ] Label as either: `automation`, `bug`, `documentation`, `enhancement`, `infrastructure`.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -207,13 +207,9 @@ RUN apk add --no-cache --virtual .perl-build-deps \
 #################
 # Install glibc #
 #################
-# Source: https://alpine-pkgs.sgerrand.com/sgerrand.rsa.pub
-# Store the key here because the above host is sometimes down, and breaks our builds
-COPY dependencies/sgerrand.rsa.pub /etc/apk/keys/sgerrand.rsa.pub
-ARG GLIBC_VERSION='2.34-r0'
 COPY scripts/install-glibc.sh /
 RUN --mount=type=secret,id=GITHUB_TOKEN /install-glibc.sh \
-    && rm -rf /install-glibc.sh /sgerrand.rsa.pub
```
Comment (ferrarimarco) [inline Dockerfile:216]: Can you please remove `dependencies/sgerrand.rsa.pub` as well?

Thanks a lot!

**Comment by ferrarimarco** [comment]
I think that this PR has a dependency on #5317, so we might have to wait for that to be merged.

File: Dockerfile (hunk 1 above)
Comment (harryzcy) [reply Dockerfile:216]: Done in fa8a53ceddd9b41c44c3f93bb62c127745d11fda

**Comment by ferrarimarco** [comment]
For the record, installing `glibc` this way was introduced in #379

**Review by ferrarimarco — APPROVED** [review]

**Pull request merged**